distributed/worker.py

```diff
@@ -4576,3 +4591,74 @@ def warn(*args, **kwargs):
         worker.log_event("warn", {"args": args, "kwargs": kwargs})
 
     warnings.warn(*args, **kwargs)
+
+
+def benchmark_disk(
+    sizes=["1 kiB", "100 kiB", "1 MiB", "10 MiB", "100 MiB"],
+    rootdir=None,
+    duration=1.0,
+) -> dict:
+    duration = parse_timedelta(duration)
+
+    out = {}
+    for size_str in sizes:
+        with tmpdir(dir=rootdir) as dir:
+            dir = pathlib.Path(dir)
+            names = list(map(str, range(100)))
+            size = parse_bytes(size_str)
+
+            data = os.urandom(size)
+
+            start = time()
+            total = 0
+            while time() < start + duration:
+                with open(dir / random.choice(names), mode="ab") as f:
+                    f.write(data)
+                    os.fsync(f)
+                total += size
+
+            out[size_str] = total / (time() - start)
+    return out
+
+
+def benchmark_memory(
+    sizes=["2 kiB", "10 kiB", "100 kiB", "1 MiB", "10 MiB"],
+    duration=0.2,
+) -> dict:
+    duration = parse_timedelta(duration)
+    out = {}
+    for size_str in sizes:
+        size = parse_bytes(size_str)
+        data = os.urandom(size)
+
+        start = time()
+        total = 0
+        while time() < start + duration:
+            data[:-1]
```

Comment:
```suggestion
            data[:]
```
this is at risk of being optimized away as a no-op, either by an alternative interpreter or by a future version of CPython

Reply:
Unfortunately this change doesn't actually force a copy

```python
In [1]: import os

In [2]: b = os.urandom(10000000)

In [3]: %timeit b[:]
40.4 ns ± 0.385 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [4]: %timeit b[:-1]
3.57 ms ± 73 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

I've added a variable assignment and deletion in order to screw with future efforts from the interpreter